_ruby_libs/pages.rb

```diff
@@ -71,6 +58,9 @@ def initialize(site, instances, repo, default)
 
     self.data['instance'] =   repo
     self.data['repo'] =   repo
+    if default then
+      self.data['redirect_from'] = [ File.join('repos', repo.name)]
```

Comment:
AFAICT, this does not work "for persisting old links" which is the stated purpose of this. Previously, the files were `_site/repos/abb/index.html`  With this new redirect, we end up with `_site/repos/abb.html`  Previously, we would be looking for http://index.ros.org/repos/abb (or `.../abb/`) Those links no longer work, instead you would have to enter http://index.ros.org/repos/abb.html to hit the redirect.

Reply:
I'm trying to merge this PR into my extended version build for index.rosdabbler.com, and I have a strange error. Github has a repo "autorobo.universe" When this PR tries to make the redirect file `repos/autorobo.universe.html`, it instead tries to write to `repos/autorobo.universe` that is without the `.html`  I assume this is because it has a '.' in the repo name. Depending on what currently exists in _site/repos, it either writes to this wrong file, or fails completely if there is an existing directory `_site/repos/autorobo.universe`  I can't see anything in your code that could cause this, so I suspect it is a bug in   `jekyll-redirect-from`  I'm not sure if this is an issue in the normal rosindex or not. I'll just work around it with a blacklist in my extended build.

Reply:
I've added this PR now to my extended index run at https://index.rosdabbler.com and I can confirm that the redirects do indeed work. I wanted to test on github pages, since that is the actual hosting site. When it did not work for me, I was using `python -m http.server -d _site` to serve. So this is dependent on the configuration of the server, but I don't think it is a problem anymore.

You've continued adding changes to this. Do you want me to look it over again?

Note as reported, I had the issue of periods in the repo name that I had to work around. Do you know if there is any ROS standard for naming of repos that does not allow periods? I could prepare you a branch that shows the issue if you want in a test run.